search-sentiments-map/test/search-test.js

```diff
@@ -1,35 +1,220 @@
+        worldData = [mockData];
+        return [worldData];
+      });      
+    })
+
+    afterEach(() => {
+      sinon.restore();
+    })
+
+    it('should retrieve the results from the datastore', async function() {
+      let results =
+          await search.retrieveSearchResultFromDatastore('topic', 'timestamp');
+      assert.deepEqual(results, mockData);
+    });
+  });
+
+  describe('RetrieveUserSearchResultFromDatastore', function() {
+    let mockData;
+    let datastoreEntities;
+
+    beforeEach(() => {
+      mockData = [
+      {
+        topic: 'testTopic',
+        timestamp: Date.now(),
+        dataByCountry: [
+          {
+            country: 'AU',
+          },
+          {
+            country: 'US',
+          }
+        ],
+      },
+      {
+        topic: 'oldTestTopic',
+        timestamp: 1593455070000,
+        dataByCountry: [
+          {
+            country: 'AU',
+          },
+          {
+            country: 'US',
+          }
+        ],
+      }];
+
+      datastoreEntities = [];
+
+      // Calls to the datastore are stubbed.
+      sinon.stub(Datastore.prototype, 'runQuery').callsFake(() => {
+        return [mockData];
+      });
+
+      // Calls to the datastore are stubbed.
+      sinon.stub(Datastore.prototype, 'key').callsFake(() => {
+        return 'fakeKey';
+      });
+
+      // Calls to the datastore are stubbed.
+      sinon.stub(Datastore.prototype, 'save').callsFake((entity) => {
+        datastoreEntities.push(entity);
+      });
+
+      // Function is stubbed because it is tested separately.
+      sinon.stub(search, 'getSearchResultsForCountriesForTopic').callsFake(
+          (countries, topic) => {
+            let data = [];
+            countries.forEach(country => {
+              data.push({country: country});
+            })
+            datastoreEntities.push(data);
+            return Promise.resolve(data);
+          });
+    })
+
+    afterEach(() => {
+      sinon.restore();
+    })
+
+    it('should retrieve search data if current data exists', async function() {
+      let results =
+          await search.retrieveUserSearchResultFromDatastore(
+              'testTopic', ['AU', 'US']);
+      assert.deepEqual(results, mockData[0]);
+    });
+
+    it('should fetch and retrieve search data if no current data exists',
+        async function() {
+          let mockCountryData = [
+            {
+              country: 'AU',
+            },
+            {
+              country: 'US',
+            }
+          ];
+          let results =
+              await search.retrieveUserSearchResultFromDatastore(
+                  'oldTestTopic', ['AU', 'US']);
+          assert.equal(results.topic, 'oldTestTopic');
+          assert.isAbove(results.timestamp, Date.now() - 60000);
+          assert.deepEqual(results.dataByCountry, mockCountryData);
+        });
+
+    it('should fetch and retrieve search data if some country data is missing',
```

Comment:
Is this statement right?
If the country data is missing it still fetches and retrieves search data?